### PR TITLE
Add Phases to Features

### PR DESCRIPTION
**Pull request by MahtabBukhari**
## Describe your changes

close: #1638

image

image

## Issue 
- Ticket Number: 1638
- [Link](https://github.com/stakwork/sphinx-tribes/issues/1638)

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Checklist before requesting a review
- [ ] I have performed a self-review of my code
- [ ] I have tested on Chrome and Firefox
- [ ] I have tested on a mobile device
- [ ] I have provided a screenshot or recording of changes in my PR if there were updates to the frontend

### CODE REVIEW

**Comment by MahtabBukhari** [comment]
@fvalentiner @elraphty Could you please reivew this PR?

File: handlers/features.go [hunk 1]
```diff
@@ -93,3 +92,71 @@ func (oh *featureHandler) GetFeatureByUuid(w http.ResponseWriter, r *http.Reques
...
+	if err != nil {
+		w.WriteHeader(http.StatusBadRequest)
+		fmt.Fprintf(w, "Error decoding request body: %v", err)
+		return
+	}
+
+	newPhase.CreatedBy = pubKeyFromAuth
+
+	phase, err := oh.db.CreateOrEditFeaturePhase(newPhase)
+	if err != nil {
```
Comment (elraphty) [inline handlers/features.go:117]: @MahtabBukhari you didn't specify the Phase Uuid which will make the code always update, when you try to create more than one phase, and also didn't account for UpdatedBY, there should be a conditional to check if CreatedBY exists, if it exists the UpdatedBy should  be updated `UpdatedBy = &now`.

Comment (MahtabBukhari) [reply handlers/features.go:117]: @elraphty Could you please review it 

image

I have updated it in conditional bases that I think good

Comment (elraphty) [reply handlers/features.go:117]: you didn't create a UUId, you to do something like this `features.Uuid = xid.New().String()`, and you don't need the updatedBy and createdBy in that condition. this works

```sh
if features.Uuid == "" {
		features.Uuid = xid.New().String()
	}
```

Comment (MahtabBukhari) [reply handlers/features.go:117]: > you didn't create a UUId, you to do something like this `features.Uuid = xid.New().String()`, and you don't need the updatedBy and createdBy in that condition. this works
> 
> ```shell
> if features.Uuid == "" {
> 		features.Uuid = xid.New().String()
> 	}
> ```

Could you please take a look it again 

image

is that's good?

File: handlers/features.go [hunk 2]
```diff
@@ -93,3 +93,79 @@ func (oh *featureHandler) GetFeatureByUuid(w http.ResponseWriter, r *http.Reques
...
+		w.WriteHeader(http.StatusBadRequest)
+		fmt.Fprintf(w, "Error decoding request body: %v", err)
+		return
+	}
+
+	if newPhase.Uuid == "" {
+		newPhase.Uuid = xid.New().String()
+	}
+
+	if newPhase.CreatedBy != "" {
```
Comment (elraphty) [inline handlers/features.go:119]: @MahtabBukhari  Make a call to the DB to get the phase, before performing the created and createdBy conditions, the createdBy won't be sent during an update, which will make that condition always true.

File: cypress/e2e/06_phases.cy.js [hunk 3]
```diff
@@ -35,7 +35,7 @@ describe('Modify phases name', () => {
                 }).its('body').then(body => {
```
Comment (elraphty) [inline cypress/e2e/06_phases.cy.js:35]: @MahtabBukhari change `06_phases.cy.js` to `cypress/e2e/06_phases.cy.ts`, everything else is fine now thanks.

**Pull request merged**